examples/cpp_execution/InferenceSetIOBufferExample.cpp

```diff
@@ -0,0 +1,511 @@
+//
+// Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+//-----------------------------------------------------------------------------
+
+#include <string>
+#include <vector>
+#include <iostream>
+#include <chrono>
+#include <algorithm>
+#include <pybind11/embed.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <optional>
+#include "/opt/qti-aic/dev/inc/QAicApi.hpp"
+#include "/opt/qti-aic/dev/inc/qaicapihpp/QAicApiDataTypes.hpp"
+
+namespace py = pybind11;
+
+namespace
+{
+
+    /**
+     * Simple helper to return true if the buffer mapping instance is an input one
+     * @param bufmap buffer mapping instance
+     * @return true if the instance is an input buffer one.
+     */
+    [[nodiscard]] bool isInputBuffer(const qaic::rt::BufferMapping &bufmap)
+    {
+        return bufmap.ioType == BUFFER_IO_TYPE_INPUT;
+    }
+
+    /**
+     * Populate input, output vectors with QBuffer information
+     * @param bufmap Buffer mapping instance
+     * @param buf Actual QBufffer that was generated at callsite/caller.
+     * @param inputBuffers Vector to use in case this is input instance
+     * @param outputBuffers Vector to use in case this is an output instance
+     */
+    void populateVector(const qaic::rt::BufferMapping &bufmap, const QBuffer &buf,
+                        std::vector<QBuffer> &inputBuffers,
+                        std::vector<QBuffer> &outputBuffers)
+    {
+
+        if (isInputBuffer(bufmap))
+        {
+            {
+                inputBuffers.push_back(buf);
+            }
+        }
+        else
+        {
+            outputBuffers.push_back(buf);
+        }
+    }
+
+    class QBufferWrapper
+    {
+    public:
+        explicit QBufferWrapper(size_t size) : buffer_{size, new uint8_t[size]} {}
+        ~QBufferWrapper() { delete[] buffer_.buf; }
+
+        [[nodiscard]] QBuffer &getQBuffer() { return buffer_; }
+
+    private:
+        QBuffer buffer_;
+    };
+    using shQBufferWrapper = std::shared_ptr<QBufferWrapper>;
+
+    [[nodiscard]] shQBufferWrapper
+    createBuffer(const std::string &bufferName,
+                 const qaic::rt::BufferMappings &allBufferMappings)
+    {
+        auto it =
+            std::find_if(allBufferMappings.begin(), allBufferMappings.end(),
+                         [&bufferName](const qaic::rt::BufferMapping &bufferMapping)
+                         {
+                             return (bufferName == bufferMapping.bufferName);
+                         });
+        if (it != allBufferMappings.end())
+        {
+            return std::make_shared<QBufferWrapper>(it->size);
+        }
+
+        throw std::runtime_error(
+            "Buffer mapping of Input Type not found for buffer named : " +
+            bufferName);
+    }
+
+    [[nodiscard]] shQBufferWrapper
+    createDecodeBuffer(const std::string &bufferName,
```

Comment:
this function is similar to createBuffer, only size is different, you can make size an extra argument to createBuffer function and remove the duplicate code.